azure-kusto-data/azure/kusto/data/_token_providers.py

```diff
@@ -370,6 +370,43 @@ def _get_token_from_cache_impl(self) -> dict:
         return self._valid_token_or_none(token)
 
 
+class InteractiveLoginTokenProvider(TokenProviderBase):
+    """ Acquire a token from MSAL with Device Login flow """
+
+    def __init__(self, kusto_uri: str, authority_uri: str, login_hint: str = None, domain_hint: str = None):
+        super().__init__(kusto_uri)
+        self._msal_client = None

```

Comment:
consider adding type hints to the class members (see my type hinting PR, maybe we want to wait for that to merge)

Reply:
Syntax supported in 3.5?

Reply:
With 3.5 you'll have to do the comment syntax, if you don't like it we can wait to 3.6

Reply:
Will wait